app/client/app/role/partials/view.html

```diff
@@ -17,6 +17,20 @@
       </div>
     </div>
   </div>
+  <div class="col-lg-12 form-group">
+    <label class="col-sm-2 control-label ng-binding">Programs</label>
+    <div class="show-value col-sm-10 col-md-8 col-lg-7">
+      <div class="ng-admin-type-string">
+        <label class="label label-default"
+               ng-if="Ctrl.getPrograms().length > 0"
```

Comment:
extract the `getPrograms()`, `hasRole()` and any other function and equate the result to a bound model... that should be used in the view. Because angularJS tends to refresh the view on every digest... thus, calling the same functions in the view multiple times... Lets talk if you need more clarity